packages/neovim.rb

```diff
@@ -3,25 +3,26 @@
 class Neovim < Package
   description 'Neovim is a refactor, and sometimes redactor, in the tradition of Vim (which itself derives from Stevie).'
   homepage 'https://neovim.io/'
-  version '0.2.0'
-  source_url 'https://github.com/neovim/neovim/archive/v0.2.0.tar.gz'
-  source_sha256 '72e263f9d23fe60403d53a52d4c95026b0be428c1b9c02b80ab55166ea3f62b5'
+  version '0.4.2'
+  source_url 'https://github.com/neovim/neovim/archive/v0.4.2.tar.gz'
+  #source_sha256 '72e263f9d23fe60403d53a52d4c95026b0be428c1b9c02b80ab55166ea3f62b5'
+  source_sha256 '9f874d3d2a74f33b931db62adebe28f8d2ec116270d1e13998b58a73348b6e56'
 
   binary_url ({
-    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/neovim-0.2.0-chromeos-armv7l.tar.xz',
-     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/neovim-0.2.0-chromeos-armv7l.tar.xz',
-       i686: 'https://dl.bintray.com/chromebrew/chromebrew/neovim-0.2.0-chromeos-i686.tar.xz',
-     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/neovim-0.2.0-chromeos-x86_64.tar.xz',
+#    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/neovim-0.2.0-chromeos-armv7l.tar.xz',
+#     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/neovim-0.2.0-chromeos-armv7l.tar.xz',
+#       i686: 'https://dl.bintray.com/chromebrew/chromebrew/neovim-0.2.0-chromeos-i686.tar.xz',
+#     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/neovim-0.2.0-chromeos-x86_64.tar.xz',
   })
   binary_sha256 ({
-    aarch64: '3dc12d8a905444dff6a70790d8b0ae27899bbb4b310f44c1d1d30e3ef9f15bcf',
-     armv7l: '3dc12d8a905444dff6a70790d8b0ae27899bbb4b310f44c1d1d30e3ef9f15bcf',
-       i686: '0ae138abca8d035eb11658d8ca8446f5c7b6c6a43c8554cdbc99c8813698223d',
-     x86_64: '16f6dfd198931f3162e1cf6054224c1e4e8059403597d621ab54b4a421fe89ed',
+#    aarch64: '3dc12d8a905444dff6a70790d8b0ae27899bbb4b310f44c1d1d30e3ef9f15bcf',
+#     armv7l: '3dc12d8a905444dff6a70790d8b0ae27899bbb4b310f44c1d1d30e3ef9f15bcf',
+#       i686: '0ae138abca8d035eb11658d8ca8446f5c7b6c6a43c8554cdbc99c8813698223d',
+#     x86_64: '16f6dfd198931f3162e1cf6054224c1e4e8059403597d621ab54b4a421fe89ed',
   })
 
   def self.build
-    system "make", "CMAKE_BUILD_TYPE=RelWithDebInfo"
+    system "make", "CMAKE_BUILD_TYPE=RelWithDebInfo", "CFLAGS=-fstack-protector-all"
```

Comment:
`"CMAKE_BUILD_TYPE=Release"` probably makes more sense since we're not building for Debian.

Reply:
I need to see the README but I'm think this might work better:
```
  def self.build
    system 'cmake', "-DCMAKE_INSTALL_PREFIX=#{CREW_PREFIX}", "-DCMAKE_BUILD_TYPE=Release"
    system 'make'
  end
```

Reply:
@uberhacker `DebInfo` stands for "Debug Info", not "Debian info".

Reply:
@uberhacker Feeling lost with `cmake` compilation since I don't find the correct CREW variable containing the tmp dir containing my `CMakelists.txt`. I also must add lines to create the `build` dir and move to it. Then I don't really understand why using cmake would work better than the command used by the neovim's developers. Could you please detail a bit more what you are expecting to improve ?

Reply:
Using this lines, I am now facing troubles with Libs paths
```
system 'mkdir build'                                                                                                                                                                                           
Dir.chdir ("build") do                                                                                                                                                                                         
    system 'cmake', "-DCMAKE_INSTALL_PREFIX=#{CREW_PREFIX}", "-DCMAKE_BUILD_TYPE=Release", ".."                                                                                                                  
    system 'make'                                                                                                                                                                                                
end 
```
which sends : 
```
CMake Error at /usr/local/share/cmake-3.15/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find LibUV (missing: LIBUV_LIBRARY LIBUV_INCLUDE_DIR) (Required
  is at least version "1.28.0")
Call Stack (most recent call first):
  /usr/local/share/cmake-3.15/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindLibUV.cmake:82 (find_package_handle_standard_args)
  CMakeLists.txt:369 (find_package)
```
Still think that 
```
def self.build
    system "make", "CMAKE_BUILD_TYPE=RelWithDebInfo", "CFLAGS=-fstack-protector-all"
end
```
is a lot easier to maintain and understand

Reply:
Not having any success building this package.  I'm not sure what changed between versions but it appears there are many more dependencies needed.

Reply:
@uberhacker sorry, very busy days this week. I am going to have a look at it.

Reply:
@uberhacker do you have the following installed ? cmake  libtool (this PR version) sha automake pkgconfig unzip wget gettext autoconf ninja

Reply:
Yeah, pretty sure I have all the dependency packages.  Most of those are already installed from Chromebrew "core".  I can try it again when I have more energy.

Reply:
That's just to be sure you have all the prerequisites. If so, I'll have a deeper look at the dependencies

Reply:
Have you tried it with `-fstack-protector-strong`? The Chrome OS compiler team created that mode specifically for Chrome OS devices to have sensible stack protection without the hit that comes from `-fstack-protector-all`.

Reply:
yes with no success on my machine. Unfortunately! My system seems to behave as if it were activated by default instead of being disabled. More infos on that [here](https://github.com/neovim/neovim/issues/11228)

Reply:
@vincowl: Please remove libtool from this PR.  It has been fixed in https://github.com/skycocker/chromebrew/pull/3566.

Reply:
I'll remove `libtool` from this PR. Please note that #3566 doesn't provide a functional `libtool` version. It needs a dependency to `flex` as described in #3537.

Reply:
@vincowl: The flex package is part of Chromebrew core so it should already be included.

Reply:
I've just made a fresh install to be sure. Indeed flex is part of the Chromebrew core but it is not installed as you try to install libtool. flex should be installed before libtool. I suppose it could be solved by changing the package list order or define flex as a libtool's dependencies.

Reply:
Here is the list that is installed : 
```
version 1.5
The following packages also need to be installed: 
binutils filecmd texinfo perl perl_locale_messages perl_text_unidecode perl_unicode_eastasianwidth ncurses gmp mpfr mpc isl cloog osl glibc gcc_tools linuxheaders make pkgconfig compressdoc doxygen readline sed acl attr gettext libxml2 zlibpkg icu4c automake autoconf m4 libsigsegv libtool intltool perl_xml_parser expat patch diffutils bison flex util_macros wget gnutls libunistring nettle openssl libressl libtasn1 trousers p11kit libffi libunbound libidn2 libpsl libmetalink gawk lzip unzip zip cmake 
```

Reply:
Yes, that is a good observation.  We shouldn't be removing dependencies of core packages.

Reply:
@uberhacker  I traced this issue in #3577 . Maybe be easier to solve it there so that we won't flood this review ?

Reply:
Since I reinstalled a fresh chromebrew environment, I can't reinstall neovim due to `libuv` compilation (at autotools configuration step). I am trying to find a solution with the neovim team, and ideally find how to use the chromebrew's `libuv` package instead of the neovim's one. Maybe the error you got @uberhacker  ?

Reply:
Fresh install problem solved in #3577 also solved my `libuv` configuration error. No trouble here (dependencies or something else) to build neovim with a brand new chromebrew installation

Reply:
@uberhacker Could you try to rebuild neovim to check if you are still unable to do so ? Good news otherwise, the neovim's team plans to use cmake and not autotools for the next releases. That would meet your initial remark linked to the use of cmake

Reply:
I modified the CMakeLists.txt file via sed commands instead of forcing compiling flags to every platform. Maybe cleaner.

Reply:
It's building for me now.